lib/jnpr/junos/device.py

```diff
@@ -730,6 +733,30 @@ def facts_refresh(self,
     def __repr__(self):
         return "Device(%s)" % self.hostname
 
+class DeviceSessionListener(SessionListener):
+
+    """
+    Listens to Session class of Netconf Transport and detects errors in the transport.
+    """
+    def __init__(self, device):
+        self._device = device
+    def callback(self, root, raw):
+        """Called when a new XML document is received. The *root* argument allows the callback to determine whether it wants to further process the document.
```

Comment:
Please run through `pep8` and clean up any reported problems like this long line.